pkg/engine/executor/project.go

```diff
@@ -41,7 +43,7 @@ func NewProjectPipeline(input Pipeline, columns []physical.ColumnExpression, eva
 			if err != nil {
 				return failureState(err)
 			}
-			fields = append(fields, arrow.Field{Name: columnNames[i], Type: vec.Type()})
+			fields = append(fields, arrow.Field{Name: columnNames[i], Type: vec.ArrowType()})
```

Comment:
Projections currently do not retain the metadata from the original fields.

Reply:
Implemented in https://github.com/grafana/loki/pull/17456/commits/5f342e671a2a37e2c8c8ad68ec7aeb057cc65815